components/builder-originsrv/src/migrations/origin_public_keys.rs

```diff
@@ -83,11 +83,15 @@ pub fn migrate(migrator: &mut Migrator) -> Result<()> {
                         RETURN;
                     END
                     $$ LANGUAGE plpgsql STABLE"#)?;
-
-    migrator.migrate("originsrv",
-                     r#"ALTER TABLE origin_public_keys
+    migrator
+        .migrate("originsrv",
+                 r#"ALTER TABLE origin_public_keys
                         DROP CONSTRAINT IF EXISTS
                           origin_public_keys_full_name_key"#)?;
-
+    migrator
```

Comment:
I think we can add a dedupe update statement that is safe to run whether there is duplicate data or not. The danger would be in a scenario where duplicate data is inserted just after the dedupe statement and before the alter statement. That could be mitigated with a transaction.

However if we are confident there is no duplicate data in production (easy to determine), then this is all unnecessary.